src/fixup.js

```diff
@@ -141,6 +141,29 @@
     }
   }
 
+  /* fragment identifiers backward compatibility mapping */
+  if (location.hash &&
+    !document.querySelector(location.hash) && // not broken link
```

Comment:
Actually, sorry... `getElementById` is safer here, because a `.hash` can contain an invalid query, which would cause an DOM exception. However, `getElementById` either matches or returns null.